File: report/src/main/java/com/vmlens/report/element/StacktraceLeaf.java
package com.vmlens.report.element;

import java.util.List;

public class StacktraceLeaf {

    private final List<StacktraceElement> stacktraceElements;

    public StacktraceLeaf(List<StacktraceElement> stacktraceElements) {
        this.stacktraceElements = stacktraceElements;
    }

    public List<StacktraceElement> stacktraceElements() {
        return stacktraceElements;
    }
}

File: sync-bug/src/main/scala/com/anarsoft/race/detection/report/element/StacktraceLeaf.scala
package com.anarsoft.race.detection.report.element

class StacktraceLeaf(val stacktraceElements: List[StacktraceElement]) {


  private def canEqual(other: Any): Boolean = other.isInstanceOf[StacktraceLeaf]
  
  override def equals(other: Any): Boolean = other match {
    case that: StacktraceLeaf =>
      that.canEqual(this) &&
        stacktraceElements == that.stacktraceElements
    case _ => false
  }
  
  override def hashCode(): Int = {
    val state = Seq(stacktraceElements)
    state.map(_.hashCode()).foldLeft(0)((a, b) => 31 * a + b)
  }
}
